### ISSUE TITLE
Arrow function componentDidMount handler breaks starting of saga

### ISSUE THREAD

**Issue by meikoudras** [keajs/kea-saga#2]
Example webpackbin [here](https://www.webpackbin.com/bins/-L-ao4r0CMSmfKNa2f-q)

When using the arrow style function for componentDidMount, the kea saga is never started.
```
componentDidMount = () => {
  console.log('component mounted');
};
```

Example shows that the start generator function is never called.

**Comment by mariusandra** [comment]
Hey, if you look at the [generated code](https://babeljs.io/repl/#?babili=false&browsers=&build=&builtIns=false&code_lz=MYGwhgzhAEDKD2BbApgFwBYEsB2BzaA3gFDTTDzYSoBOArsKvNdABQCUhJpZFE8IyAHQh4uFgHIqYaqhy5xbLgF8uXKdgAm0jQDFa2BpgqsOxbj0r8hIseIBG4BcqJdp1eAHc9B2cYC8JtB-AHyc5uSWAsKiEg5gcU6kKipEEXxRNiwIKBhyggAO7oyoAJ75Qupa1Lr6hhSKAPQN0AC0oQBmtb7YgQQpaVbRYtloWHgFRfCl5YJunt512I3NbdD6GsjtOMgaLgKo0Hw5Y_gB2MgecEijcuxERzfjldoL3XdEQA&debug=false&circleciRepo=&evaluate=true&lineWrap=false&presets=es2015%2Creact%2Cstage-2&prettier=false&targets=&version=6.26.0) for arrow functions:

<img width="1603" alt="screen shot 2017-12-20 at 14 22 59" src="https://user-images.githubusercontent.com/53387/34209176-6e10e608-e591-11e7-9a2e-92d6a218aa68.png">

... then while regular functions are added to the "class"'s `prototype` and can be accessed before the instance is created, declaring your functions in the arrow style only adds them to the class's instance in the transpiled constructor.

`kea-saga` takes your component's `componentDidMount` and wraps it with the code that starts the sagas. If you write your `componentDidMount` with an arrow function, then whatever `kea-saga` injects will be overwritten when an instance of your class is created.

Thus there's no good solution here.

However, I don't actually see a reason to have `componentDidMount` defined as an arrow function. It should only be called once in your component's lifecycle and then `this` will be a reference your component's instance.

In any case, thanks for the find. I have added a notice about it to [the documentation](https://kea.js.org/effects/saga). :)

**Comment by mariusandra** [comment]
This should be fixed in 1.0
